File: CS/Creational/Factory/FactoryPatternRunner.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Pattern.Creational.Factory
{
	internal static class FactoryPatternRunner
	{
		public static void Run()
		{
			Console.WriteLine("Factory Pattern Runner");

			var products = new List<IProduct>();

			for(int i = 0;i<10; i++)
			{
				var productType = (new Random().Next(0,2) < 1) ? "A" : "B";
				var product = new ProductFactory().Create(productType);
				products.Add(product);
			}

			foreach(var product in products)
			{
				Console.WriteLine(product.ToString());
			}
		}
	}
}

File: VB/Creational/Factory/FactoryPatternRunner.vb
Imports System.Text

Public Class FactoryPatternRunner

	Public Sub Run()

		Console.WriteLine("Factory Pattern..")

		Dim products = New List(Of IProduct)
		Dim factory As New Factory()
		Dim random As New Random()


		For i As Integer = 0 To 10
			Dim type As String = ""
			If random.NextDouble() < 0.5 Then
				type = "A"
			Else
				type = "B"
			End If

			Dim product As IProduct = factory.CreateProduct(type)
			If product IsNot Nothing Then
				products.Add(product)
			End If
		Next

		Dim sbProductTypes As New StringBuilder
		For Each product In products
			sbProductTypes.Append(product.ToString)
		Next

		Console.WriteLine("Products factoried: {0}", sbProductTypes.ToString)

	End Sub

End Class
